src/web/rustdoc.rs

```diff
@@ -483,6 +484,24 @@ pub fn rustdoc_html_server_handler(req: &mut Request) -> IronResult<Response> {
 
     let latest_path = format!("/crate/{}/latest{}{}", name, target_redirect, query_string);
 
+    let canonical_url = if krate.documentation_url.is_none()
+        || krate
+            .documentation_url
+            .as_ref()
+            .unwrap()
+            .starts_with("https://docs.rs/")
+    {
+        format!(
+            "{}://{}/{}/latest/{}",
+            req.url.scheme(),
+            req.url.host(),
```

Comment:
canonical URLs can be anything that a link accepts, so we also could just omit everything except the path

Reply:
According to https://developers.google.com/search/docs/advanced/crawling/consolidate-duplicate-urls:

> Use absolute paths rather than relative paths with the rel="canonical" link tag.
> Recommended: https://www.example.com/dresses/green/greendress.html
> Not recommended: /dresses/green/greendress.html

Reply:
ok, thanks for the info, didn't know that!

Reply:
not sure I heard an answer here - @jsha why did you choose to use `req.url.host()`?